samples/samples/snippets.py

```diff
@@ -3017,6 +3017,51 @@ def directed_read_options(
     # [END spanner_directed_read]
 
 
+def set_custom_timeout_and_retry(instance_id, database_id):
+    """Executes a snapshot read with custom timeout and retry."""
+    # [START spanner_set_custom_timeout_and_retry]
+    from google.api_core import retry
+    from google.api_core import exceptions as core_exceptions
+
+    # instance_id = "your-spanner-instance"
+    # database_id = "your-spanner-db-id"
+    spanner_client = spanner.Client()
+    instance = spanner_client.instance(instance_id)
+    database = instance.database(database_id)
+
+    retry = retry.Retry(
+        # Customize retry with an initial wait time of 500 milliseconds.
+        initial=0.5,
+        # Customize retry with a maximum wait time of 16 seconds.
+        maximum=16,
+        # Customize retry with a wait time multiplier per iteration of 1.5.
+        multiplier=1.5,
+        # Customize retry with a timeout on
+        # how long a certain RPC may be retried in
+        # case the server returns an error.
+        timeout=60,
+        # Configure which errors should be retried.
+        predicate=retry.if_exception_type(
+            core_exceptions.ServiceUnavailable,
```

Comment:
https://github.com/googleapis/python-api-core/blob/4fed37cbc32122f156e38250b5fa8b2b08a787a1/google/api_core/exceptions.py#L420

ServiceUnavailable is same as gRPC `UNAVAILABLE` error code.